outport/process/outportDataProvider.go

```diff
@@ -134,6 +139,9 @@ func (odp *outportDataProvider) PrepareOutportSaveBlockData(arg ArgPrepareOutpor
 		return nil, err
 	}
 
+	stateChanges := odp.stateChangesCollector.GetStateChangesForTxs()
+	odp.stateChangesCollector.Reset()
```

Comment:
I would move the reset on the commit block, after the indexing part. 
Otherwise, in case there is no client for the outport block, you would not end up in this method and the reset would not be done.